STX.SPAL.Core/Services/Foundations/Assemblies/AssemblyService.Exceptions.cs

```diff
@@ -0,0 +1,36 @@
+﻿// ----------------------------------------------------------------------------------
+// Copyright (c) The Standard Organization: A coalition of the Good-Hearted Engineers
+// ----------------------------------------------------------------------------------
+
+using System.Reflection;
+using STX.SPAL.Core.Models.Services.Foundations.Assemblies.Exceptions;
+using Xeptions;
+
+namespace STX.SPAL.Core.Services.Foundations.Assemblies
+{
+    internal partial class AssemblyService
+    {
+        private delegate Assembly ReturningAssemblyFunction();
+
+        private static Assembly TryCatch(ReturningAssemblyFunction returningAssemblyFunction)
+        {
+            try
+            {
+                return returningAssemblyFunction();
+            }
+
+            catch (InvalidAssemblyPathException invalidAssemblyPathException)
+            {
+                throw CreateAssemblyValidationException(
+                    invalidAssemblyPathException);
+            }
```

Comment:
Looks good, you just need to handle the external exceptions and service exception

Dependency Validation Exceptions: ArgumentException
Dependency Exceptions: InvalidOperationException, NotSupportedException, SecurityException, IOException, BadImageFormatException, UnauthorizedAccessException

Reply:
![image](https://github.com/The-Standard-Organization/STX.SPAL.Core/assets/44029921/90380a3b-3f45-4e0a-a5d8-2deb17b4a80c)
![image](https://github.com/The-Standard-Organization/STX.SPAL.Core/assets/44029921/a0fcac9b-0260-40ea-b601-4640ddbcf5d1)